### ISSUE TITLE
Add glibc version constraints to kubelet 1.29 package

### ISSUE THREAD

**Issue by saschagrunert** [kubernetes/release#3246]
https://github.com/kubernetes/release/issues/3076 will land in Kubernetes 1.29, containing a switch to debian:bookworm. This means we now require glibc 2.36 (compared to 2.31 on bullseye) for non static binaries like the kubelet.

The package of the kubelt could contain a constraints like this to highlight possible incompatibilities. 

```
Requires: glibc >= 2.36
```

This needs to be done per release branch, while only affecting 1.29. We could also add a similar constraint to previous release branches (with 2.31).

cc @kubernetes/release-managers

**Comment by cpanato** [comment]
We might upgrade the active branches to 1.21 as well so if that happens we need that constrain in the others as well

**Comment by saschagrunert** [comment]
We'll bump back to bullseye with https://github.com/kubernetes/release/pull/3249, means we can add the `2.31` glibc constraint to all maintained release branches.

**Comment by dweineha** [comment]
Current Ubuntu LTS (22.04) uses 2.35 (and the next version won't release until 5 months after Kubernetes 1.29). Am I understanding things correctly that Kubernetes 1.29 won't be usable on Ubuntu 22.04 LTS?

**Comment by xmudrii** [comment]
/remove-lifecycle stale
/retitle Add glibc version constraints to the kubelet package

**Comment by cpanato** [comment]
We need to get back to this to be able to update the images

How can we make this move forward?

cc @saschagrunert @xmudrii @dims @liggitt @BenTheElder

**Comment by xmudrii** [comment]
@cpanato I would propose creating a new issue in k/k to discuss how do we want to handle the glibc upgrade. Handling it in the kubelet package should be straightforward, but we need to communicate this properly and consider what are our options.

**Comment by BenTheElder** [comment]
I guess the tricky part is aligning this project-wide and ensuring we have per-release configuration for the specs.

The latter is something we need for other fixes to these packages anyhow still, if I remember correctly?

For the former, we could perhaps derive it from kube-cross? But if we want to define a policy otherwise we broadly need a broader, higher-vis discussion.

**Comment by xmudrii** [comment]
> I guess the tricky part is aligning this project-wide and ensuring we have per-release configuration for the specs.

This is not a problem with the new infrastructure and tooling, we can do it easily, we just didn't need to do it at all. We can also have different dependencies and versions without dedicated per-release specs, e.g. https://github.com/kubernetes/release/blob/master/cmd/krel/templates/latest/metadata.yaml

However, we need policies around glibc version that we use, this is indeed project-wide decision. Right now we don't have it, but we'll need to come up with something soon.
